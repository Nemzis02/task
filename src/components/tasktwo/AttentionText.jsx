import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import TriangleIcon from './TriangleIcon';

const Row = styled.div`
  display: flex;
  align-items: center;
  width: 100%;
  border-bottom: 1px solid #d1d1d1;
`;

const Text = styled.span`
  margin-left: 10px;
  font-size: 20px;
  font-weight: bold;
`;

const AttentionText = ({ text }) => (
  <Row>
    <TriangleIcon />
    <Text>{text}</Text>
  </Row>
);

AttentionText.propTypes = {
  text: PropTypes.string.isRequired,
};

export default AttentionText;
