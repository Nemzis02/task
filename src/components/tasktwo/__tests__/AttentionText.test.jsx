import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import AttentionText from '../AttentionText';

describe('AttentionText Component', () => {
  it('Should be rendered correctly', () => {
    const { container } = render(<AttentionText text="Some text" />);
    expect(container).toMatchSnapshot();
  });

  test('Should render passed through props text', () => {
    const text = 'Test text';

    render(<AttentionText text={text} />);
    const result = screen.getByText(text);
    expect(result).toBeInTheDocument();
  });

  test('Should render warning icon', () => {
    render(<AttentionText text="Any text" />);
    const result = screen.getByAltText('Triangle');

    expect(result).toBeInTheDocument();
  });
});
