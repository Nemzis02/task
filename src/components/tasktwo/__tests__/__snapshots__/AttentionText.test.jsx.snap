// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AttentionText Component Should be rendered correctly 1`] = `
<div>
  <div
    class="AttentionText__Row-sc-1h0csey-0 djJXsm"
  >
    <div
      class="TriangleIcon__Container-kelki1-0 dBbNcD"
    >
      <div
        class="TriangleIcon__Triangle-kelki1-2 iABfzB"
      />
      <img
        alt="Triangle"
        class="TriangleIcon__IconImage-kelki1-1 fcbLiJ"
        src="images/triangle.svg"
      />
    </div>
    <span
      class="AttentionText__Text-sc-1h0csey-1 khFFMU"
    >
      Some text
    </span>
  </div>
</div>
`;
